### PR TITLE
Speedup latch sync a bit

### PR DESCRIPTION
**Pull request by AidanAbd**

### CODE REVIEW

File: latch_cli/services/sync.py [hunk 1]
```diff
@@ -345,4 +353,7 @@ def sync(
             )
         click.echo()
 
-    sync_rec(srcs, dest, delete=delete)
+    with ProcessPoolExecutor(max_workers=get_max_workers()) as executor:
+        futures = sync_rec(srcs, normalize_path(dest), delete=delete, executor=executor)
+        for future in as_completed(futures):
+            future.result()
```
Comment (ayushkamat) [inline latch_cli/services/sync.py:359]:
```suggestion
        sync_rec(srcs, normalize_path(dest), delete=delete, executor=executor)
```

`executor.__exit__()` will act like calling `executor.shutdown(wait=True)` which will wait for all the outstanding futures in the executor (see https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown)

Comment (ayushkamat) [reply latch_cli/services/sync.py:359]: this way you also don't have to change the return type of the fn

**Review by ayushkamat — CHANGES_REQUESTED** [review]
fine other than one comment

we should probably do one big gql at the start and then multiprocess the downloads but w/e

File: latch_cli/services/sync.py (hunk 1 above)
Comment (AidanAbd) [reply latch_cli/services/sync.py:359]: Goo comment.

**Pull request merged**